flutter-example/lib/src/sample_feature/sample_item_list_view.dart

```diff
@@ -1,253 +1,150 @@
+    final response =
+        await _webViewKey.currentState?.sendRpcMessage("getCredentials", {});
 
-    final assetManifest = await rootBundle.loadString('AssetManifest.json');
-    final assets = Map<String, dynamic>.from(json.decode(assetManifest));
+    // Parse the credentials
+    final credentials = response?['body']?['result'] ?? [];
 
-    final List<String> files =
-        assets.keys.where((key) => key.startsWith('assets/dock-sdk/')).toList();
-
-    for (final asset in files) {
-      final data = await rootBundle.load(asset);
-      final bytes = data.buffer.asUint8List();
-      final file = File('${targetDir.path}/${asset.split('/').last}');
-      await file.writeAsBytes(bytes, flush: true);
-    }
-
-    final indexPath = '${targetDir.path}/index.html';
-    _controller.loadFile(indexPath);
-    // Every update made in the webivew code, requires a rebuild in the flutter app
-    // To improve the development experience, you can load the webivew code from a local server
-    // Uncomment the line below and replace the IP address with your local IP address
-    // Make sure to run the local server before uncommenting the line below
-    // _controller.loadRequest(Uri.parse("http://192.168.1.117:8080/index.html"));
+    // Update state with parsed credentials
+    setState(() {
+      _credentials = credentials;
+      _isLoading = false;
+    });
   }
 
-  String _generateJsonRpcRequest(
-      String method, Map<String, dynamic> args, int id) {
-    final request = {
-      "type": "json-rpc-request",
-      "body": {
-        "jsonrpc": "2.0",
-        "method": method,
-        "params": {
-          "__args": [args]
-        },
-        "id": id
-      }
-    };
-    return jsonEncode(request);
-  }
+  Future<void> _clearData() async {
+    // Clear credentials data using `clearData` method
+    await _webViewKey.currentState?.sendRpcMessage("clearData", {});
 
-  void sendRpcMessage(String method, Map<String, dynamic> args, int id) {
-    _rpcRequestStartTime = DateTime.now(); // Start timing the RPC request
-    final jsonRpcRequest = _generateJsonRpcRequest(method, args, id);
-    final script = 'window.postMessage(${jsonEncode(jsonRpcRequest)}, "*");';
-    _controller.runJavaScript(script);
+    // Update UI after clearing data
+    setState(() {
+      _credentials = []; // Clear credentials list in UI
+      _isLoading = false;
+    });
   }
 
-  @override
-  Widget build(BuildContext context) {
-    return WebViewWidget(controller: _controller);
-  }
-}
-
-class SampleItemListView extends StatefulWidget {
-  const SampleItemListView({
-    super.key,
-    this.items = const [SampleItem(1), SampleItem(2), SampleItem(3)],
-  });
-
-  static const routeName = '/';
-
-  final List<SampleItem> items;
-
-  @override
-  _SampleItemListViewState createState() => _SampleItemListViewState();
-}
+  Future<void> _importCredential() async {
+    setState(() {
+      _isLoading = true;
+    });
+
+    const qrCodeData =
```

Comment:
add a comment here telling them they'll need to get this URL from a QR code scan